src/index.js

```diff
@@ -140,7 +140,15 @@ module.exports = Class.extend({
       distConfig = dist.Properties.DistributionConfig;
 
       if (pathPattern) {
-         cacheBehavior = _.findWhere(distConfig.CacheBehaviors, { PathPattern: pathPattern });
+         // If an object is being passed in let's assume it's a reference
```

Comment:
I sincerely apologize for this taking so long. I keep getting caught up with other PRs.

This could potentially be other objects, but I'm not personally sure what they all are. This will add functionality for at least the Reference and we can always add more to it if we find that other objects want to be used. That being said...

>  "if this is an object, find the cache behavior that has a deeply-equal object"?

Can you elaborate on this a bit? If there's a better option here I'd love to implement it, but I'm not quite sure what you're suggesting here. Thanks!